mesh/meshdb.go

```diff
@@ -362,6 +363,18 @@ func getRewardKeyPrefix(account types.Address) []byte {
 	return []byte(str)
 }
 
+// adding functions for getting the reward key for a smesherID
+// format for the index "reward_<smesherid>_<layerid> -> reward_<accountid>_<layerid> -> the actual reward"
+func getSmesherRewardKey(l types.LayerID, smesherID types.NodeID) []byte {
+	str := string(getSmesherRewardKeyPrefix(smesherID)) + "_" + strconv.FormatUint(l.Uint64(), 10)
```

Comment:
I think you need to account for endianness here to ensure sort works across platforms, using `binary.BigEndian.PutUint*`. See Noam's comment here: https://github.com/spacemeshos/go-spacemesh/pull/2176#issuecomment-766951415

Reply:
Currently the key is stored as a byte representation of a string where the layerId is being written as a string representation of a base 10 number. How does endianness come into play here, since each integer character will take only 1 byte here.